tests/test_saltworker.py

```diff
@@ -53,3 +62,236 @@ def test_valid_environment(saltworker_client):
+
+    saltworker_win = SaltWindows(system_params, **salt_config)
+
+    saltworker_win._prepare_for_install = MagicMock(return_value=None)
+    saltworker_win._install_package = MagicMock(return_value=None)
+    saltworker_win.service_stop = MagicMock(return_value=None)
+    saltworker_win._build_salt_formula = MagicMock(return_value=None)
+    saltworker_win.service_disable = MagicMock(return_value=True)
+    saltworker_win._set_grain = MagicMock(return_value=None)
+    saltworker_win.process_grains = MagicMock(return_value=None)
+    saltworker_win.run_salt = MagicMock(return_value=None)
+    saltworker_win.working_dir = system_params['workingdir']
+    saltworker_win.cleanup = MagicMock(return_value=None)
+
+    saltworker_win.install()
+
+    # assertions ===================
+    assert saltworker_win._prepare_for_install.call_count == 1
+    assert saltworker_win._install_package.call_count == 1
+    saltworker_win.service_stop.assert_called_with('salt-minion')
+    saltworker_win._build_salt_formula.assert_called_with(
+        saltworker_win.salt_srv
+    )
+    saltworker_win.service_disable.assert_called_with('salt-minion')
+    saltworker_win._set_grain.assert_called_with(
+        'ash-windows', {'lookup': {'role': salt_config['ash_role']}}
+    )
+    assert saltworker_win.process_grains.call_count == 1
+    saltworker_win.run_salt.assert_called_with('pkg.refresh_db')
+    assert saltworker_win.cleanup.call_count == 1
+
+
+@patch.dict(os.environ, {'systemdrive': 'C:'})
+@patch('codecs.open', autospec=True)
+@patch('os.makedirs', autospec=True)
+@patch('yaml.safe_dump', autospec=True)
+def test_windows_prep_install(mock_safe, mock_makedirs, mock_codec):
+    """Ensure that prep portion of install runs as expected."""
+    system_params = {}
+    salt_config = {}
+    system_params['prepdir'] = "ac2bf0c3-7985-569f-bfbd-3a8d8a13ce7d"
+    system_params['logdir'] = "5b0976f8-fcbc-50af-9459-8060589e70d9"
+    system_params['workingdir'] = "860f630a-f85d-5ed2-bd7a-bbdb48a53b2b"
+
+    salt_config['installer_url'] = "5f0c8635-4a10-5802-8145-732052a0b44b"
+    salt_config['ash_role'] = "fddc3dc3-3684-5924-bf55-bb1dbc4e4c08"
+
+    saltworker_win = SaltWindows(system_params, **salt_config)
+
+    saltworker_win.create_working_dir = MagicMock(
+        return_value=system_params['workingdir']
+    )
+    saltworker_win._prepare_for_install()
+
+    # assertions ===================
+    saltworker_win.create_working_dir.assert_called_with(
+        system_params['workingdir'],
+        'Salt-'
+    )
+    mock_makedirs.assert_called_with(saltworker_win.salt_conf_path)
+    mock_codec.assert_called_with(
+        os.path.join(saltworker_win.salt_conf_path, 'minion'),
+        'w',
+        encoding="utf-8"
+    )
+    mock_safe.assert_called_with(
+        saltworker_win.salt_conf,
+        mock_codec.return_value.__enter__(),
+        default_flow_style=False
+    )
+
+
+@patch('os.path.exists', autospec=True)
```

Comment:
These tests of watchmaker.utils should probably go in another file, `test_utils.py`